contracts/bonding/BondingVotes.sol

```diff
@@ -461,17 +465,20 @@ contract BondingVotes is ManagerProxyTarget, IBondingVotes {
         view
         returns (uint256)
     {
-        EarningsPool.Data memory startPool = getTranscoderEarningsPoolForRound(
-            bond.delegateAddress,
-            bond.lastClaimRound
-        );
+        address transcoder = bond.delegateAddress;
+        EarningsPool.Data memory startPool = getTranscoderEarningsPoolForRound(transcoder, bond.lastClaimRound);
 
-        (uint256 rewardRound, EarningsPool.Data memory endPool) = getLastTranscoderRewardsEarningsPool(
-            bond.delegateAddress,
-            _round
-        );
+        (
+            BondingCheckpoint storage transcoderBond,
+            EarningsPool.Data memory endPool
+        ) = getLastTranscoderRewardsEarningsPool(transcoder, _round);
+
+        if (!isRegisteredTranscoder(transcoder, transcoderBond)) {
+            // Delegating to an account that is not actually a transcoder should render no voting power.
```

Comment:
IIUC the `delegatorCumulativeStakeAt()` function is now responsible for the following:

- If the delegator is NOT delegated to a registered transcoder as of the snapshot round, return 0 voting power
- If the delegator is delegated to a registered transcoder as of the snapshot round, return the delegator's pending stake as of the snapshot round

Given this, would a more descriptive name for this function be `delegatorVotesAt()` since the function returns voting power which depends on the delegator's pending stake, but is not necessarily equal to the delegator's pending stake?

If you agree with this, I also suggest updating the comments to reflect this.

Reply:
Yeah I agree that the implementation here is diverging from simply returning "checkpointed stake". I'm ok with the renaming, tho a weird bit to me would be the `getBondingState` function (which doesn't "know" votes, it comes from that previous implementation of `BondingCheckpoints`) calling a function `delegatorVotesAt`. We could rename `getBondingState` to something else as well (any ideas?), but keep in mind there would be a lot of changes since its a public function used in tests etc (OTOH if the "bonding state" name seems inaccurate as well, we probably want to rename it and reduce the confusion)

One alternative could be moving the `transcoderBond` variable creation and `isRegisteredTranscoder` check to the `getBondingState` function. That might need some moving around tho but could be neat. Then it passes both `bond` and `transcoderBond` down to `delegatorCumulativeStakeAt`, 

Let me know which path you prefer!

Reply:
What about `getBondingState()` -> `getVotingState()`?

> but keep in mind there would be a lot of changes since its a public function used in tests

IMO if the changes are primarily in tests then that seems like a reasonable cost (assuming the cost is just a find+replace & bigger diff in this PR) to pay for clearer naming.

Reply:
OK I went with:
- `getBondingStateAt` -> `getVotesAndDelegateAtRoundStart`
- `delegatorCumulativeStakeAt` -> `delegatorVotesAtRoundStart`

Even tho the names are a bit long, I wanted to be extra specific there because there's another key difference: these functions return the state at the **start** of the round while the ERC-5805 ones return at the _end_. So this doesn't return the same "votes" as the other functions. I also cleaned up some documentation about it, moving the disclaimers about this difference to these functions, instead of repeating them on all the ERC-5805 functions.

> IMO if the changes are primarily in tests then that seems like a reasonable cost (assuming the cost is just a find+replace & bigger diff in this PR) to pay for clearer naming.

Restricted the changes to function&variable names, documentation and tests. Self-reviewed a couple of times as well and it looks right to me.